src/EcomDestinationWriter.cs

```diff
@@ -1788,6 +1810,43 @@ private void WritePrices(Dictionary<string, object> row, Dictionary<string, Colu
         }
     }
 
+    private void WriteUnitOfMeasure(Dictionary<string, object> row, Dictionary<string, ColumnMapping> columnMappings, DataRow dataRow)
+    {
+        bool didFindUnitOfMeasure = false;
+        if (columnMappings.TryGetValue("UnitOfMeasureCode", out var unitOfMeasureCodeColumn))
+        {
+            var unitOfMeasureCodeLookupValue = GetMergedValue(unitOfMeasureCodeColumn, row);
+            if (!string.IsNullOrWhiteSpace(unitOfMeasureCodeLookupValue))
+            {
+                var unitOfMeasureIDs = new List<string>();
+                unitOfMeasureIDs = ExistingUnitOfMeasures.Select("UnitOfMeasureCode='" + unitOfMeasureCodeLookupValue + "'").Select(r => Converter.ToString(r["UnitOfMeasureId"])).ToList();
+                if (unitOfMeasureIDs.Count > 0)
+                {
+                    dataRow["UnitOfMeasureId"] = unitOfMeasureIDs[0];
+                    didFindUnitOfMeasure = true;
+                }
+            }
+        }
+        if (!didFindUnitOfMeasure && columnMappings.TryGetValue("UnitOfMeasureExternalId", out var unitOfMeasureExternalIdColumn))
+        {
+            var unitOfMeasureExternalIdLookupValue = GetMergedValue(unitOfMeasureExternalIdColumn, row);
+            if (!string.IsNullOrWhiteSpace(unitOfMeasureExternalIdLookupValue))
+            {
+                var unitOfMeasureIDs = new List<string>();
+                unitOfMeasureIDs = ExistingUnitOfMeasures.Select("UnitOfMeasureExternalId='" + unitOfMeasureExternalIdLookupValue + "'").Select(r => Converter.ToString(r["UnitOfMeasureId"])).ToList();
+                if (unitOfMeasureIDs.Count > 0)
+                {
+                    dataRow["UnitOfMeasureId"] = unitOfMeasureIDs[0];
+                }
+            }
+        }
+
+        if (!columnMappings.TryGetValue("UnitOfMeasureLanguageId", out _))
+        {
+            dataRow["UnitOfMeasureLanguageId"] = Ecommerce.Services.Languages.GetDefaultLanguageId();
```

Comment:
use _defaultLanguageId class variable instead of service call

Reply:
Sure, i did not see that.. thx :)